server/src/main/java/org/candlepin/controller/ContentAccessManager.java

```diff
@@ -322,7 +321,8 @@ public ContentAccessCertificate getCertificate(Consumer consumer)
         if (oeca == null || regenerate) {
             log.info("Generating new SCA payload for organization \"{}\"...", owner.getKey());
 
-            String contentJson = this.createPayloadAndSignature(owner, env);
+            byte[] payload = this.createContentAccessDataPayload(owner, env, consumer);
```

Comment:
This will cause the content payload to be generated for the owner-env-consumer triplet, but the OwnerEnvContentAccess object is stored only per-owner-environment. I suspect this will lead to a similar problem where content is either missing or extraneous depending on which consumer triggers cert generation first.